src/types/bytes.rs

```diff
@@ -54,7 +54,7 @@ impl<'a> Visitor<'a> for BytesVisitor {
     where
         E: Error,
     {
-        if value.len() >= 2 && &value[0..2] == "0x" {
+        if value.is_ascii() && value.len() >= 2 && &value[0..2] == "0x" {
```

Comment:
```suggestion
        if value.starts_with("0x") {
```

Wouldn't actually that work better?

Reply:
Oh, indeed! However, it can be expressed even better with `str::strip_prefix`. Pushed the change.